### DISCUSSION TITLE
Raspberry Pi

### DISCUSSION THREAD

**Discussion by Maximal30** [n3d1117/chatgpt-telegram-bot#52] [Q&A] [answered]
Hello. Great job! Did you manage to run the program on raspberry py? I'm not a programmer and I'm having some dependency issues. On the main pc launched without problems. Can you advise how you can run this on a raspberry?

**Comment by n3d1117** [comment, marked as answer]
Hello @Maximal30, I don't have a raspberry pi. What error are you facing there?

**Reply by Maximal30** [reply]
Thank you for your response, I have figured it out already. I had RaspberyOS x64 installed, but after reinstalling Ubuntu Server LTS x64 and installing the necessary packages, everything started working! Thank you very much for your work!

**Comment by AlejandroIglesiasCalvo** [comment]
I was able to complete the task, but encountered an issue with tiktoken. It does not support armv7 wheels, which is required for Raspberry's OS. To overcome this, I created a Buildx group of docker. The Windows version works perfectly, but the arm version is built without statistics. I tried finding a workaround, but was unsuccessful. Although the bot functions properly, it always returns 1 when asked for statistics. It would be helpful to use another token count library that is compatible with arm processors.
